Updating trace view page for completeness and accuracy and graph view-DOCS-1903 (#2029)
wandb/docs: weave/guides/tracking/trace-tree.mdx

---
title: "Navigate the Weave Trace view"
description: "Use Weave's Trace view to make sense of complex execution paths in your LLM and agentic apps."
---

The Weave Trace view is designed to help you make sense of complex execution paths in your LLM and agentic apps. Whether you're debugging an agentic app with dozens of nested calls, or tracking the flow of a single model prediction, the Trace view provides a clear visualization of what happened. It also provides alternate ways to view and understand your application flow.

This guide describes how to move through the trace stack, filter and search for functions called by your code, switch between visual representations, and more.

## Get started

To enter the Trace view:
1. Navigate to [https://wandb.ai](https://wandb.ai) and select your project.
2. In the sidebar menu, choose **Traces** to view all traces saved for your project.
3. Choose a trace to open the Trace Details view. The Trace Details view displays additional panels with a hierarchical breakdown of the trace execution.

## Traces page overview

The Traces page is composed of three core panels:

> - _Left panel_: A sortable, paginated list of all trace runs for the project.

Reviewer: I'm not sure these should be italicized. Typically, we use bold here on bulleted lists. Google style guide seems to agree

  - This traces table includes additional data such as tokens, cost, and latency.
- _Center panel_: Interactive trace view for a selected trace. The trace tree shows a hierarchy of all the methods tracked within the trace. 
  - The trace tree displays ops, which are `@weave.op()`-decorated functions, that were called during the trace. 

>   - A Weave op is a versioned function that automatically logs all calls.

Reviewer: I think the wording on this is kinda confusing, but I also don't think this line is necessary. IMO, we should nix it.

- _Right panel_: Details for a selected op within the selected trace.

<Frame>
![Traces page showing a selected trace and selected op details ](/weave/guides/tracking/imgs/trace-tree-full.png)
</Frame>

### Filter within a trace

> - _Regex filter by name(s)_: Use the text field above the trace tree to filter ops by name or type, such as `tool`, `openai.response.create`.

Reviewer: same comment as above

- _Metrics_: Control whether to display the following data metrics when available: cost, tokens, and latency.

### Navigate a trace
The trace tree shows a hierarchy of all the methods tracked within the trace. To move up/down the tree, use `Cmd` (macOS) or `Alt` (Windows/Linux) + Up Arrow (↑) / Down Arrow (↓).

> There are several scrubbers available below the trace tree.  These scrubbers provide rapid navigation across states within the trace tree.  You can use the sliders to navigate strategically through your trace.

Reviewer: ```suggestion
There are several scrubbers below the trace tree that provide rapid navigation across states within it.  You can use the sliders to strategically navigate through your trace.
```
This reads cleaner to me

Expand the panel to see all available scrubbers:
- **Timeline**: Chronological order of events within the trace.
- **Peers**: Ops sharing the same type. For example, if you were examining details of a function called `predict`, you can use this scrubber to immediately jump to the next execution of `predict` within the trace.

> - **Siblings**: Ops with the same parent.  Use this scrubber to iterate over ops nested under the parent function call.

Reviewer: all of this is good stuff

<Frame>
![Trace view showing the code view of a trace](/weave/guides/tracking/imgs/trace-tree-code-view.png)
</Frame>

#### Flame graph

The flame graph view provides a timeline-based visualization of execution depth and duration. This is helpful for when trying to understand performance diagnostics over time. You can select into frames to isolate sub-traces.

<Frame>
![Trace view showing a flame graph of ops within a trace](/weave/guides/tracking/imgs/trace-tree-flame-view.png)
</Frame>

#### Graph view

The graph view shows hierarchical relationships between ops. This is useful for understanding parent/child relationships.

<Frame>
![Trace view showing a functional graph view of ops and relationships within a trace](/weave/guides/tracking/imgs/trace-tree-graph-view.png)
</Frame>

### View details for a trace op

After you have selected an op in the trace tree, details for that op display in the next panel. These details are grouped into the following tabs:

- **Call**: The input and output to the op execution.
- **Code**: The code that was used when the call was made.

> - **Feedback**: Any available feedback for the op. You can provide feedback directly within Weave or through the API.

Reviewer: might be worth linking to the feedback doc here: https://docs.wandb.ai/weave/guides/tracking/feedback

> - **Scores**: Any available scores for the op. Calls are scored by running Evaluations.

Reviewer: same for scores
